docs/plugins.md

```diff
@@ -60,3 +60,15 @@ You need to register a `plugin` only once per JSS instance. There is a number of
   })
   ```
 
+1. Hook `onProcessSheet(sheet)`.
+
+  This hook is invoked on every created `StyleSheet` after all rules are processed, with the `sheet` as an argument.
+
+  ```javascript
+  jss.use({
+    onProcessSheet: (sheet) => {
+      // Do something here.
```

Comment:
Well in theory I am fully for immutable style, though in case of high performance cases like rule plugins, we do modify objects for the sake of performance, at some point we may think how to make it immutable but still not loose any perf.